src/utils/src/files_functions.rs

```diff
@@ -0,0 +1,242 @@
+use std::{path::{PathBuf, Path}, io::Cursor, ffi::OsStr};
+
+use dioxus::desktop::DesktopContext;
+use futures::StreamExt;
+use mime::*;
+use tokio_util::io::ReaderStream;
+use warp::{constellation::{Progression, directory::Directory}, error::Error};
+use image::io::Reader as ImageReader;
+
+use crate::Storage;
+
+pub async fn upload_file(file_storage: Storage, file_path: PathBuf, eval_script: DesktopContext) {
+    let mut filename = match file_path
+        .file_name()
+        .map(|file| file.to_string_lossy().to_string())
+    {
+        Some(file) => file,
+        None => return,
+    };
+
+    let local_path = Path::new(&file_path).to_string_lossy().to_string();
+    let mut file_storage = file_storage.clone();
```

Comment:
```suggestion
```

This would be because `file_storage` is mutable above so no need to clone it here.